### ISSUE TITLE
Sitemap.xml - add and configure.

### ISSUE THREAD

**Issue by finnlewis** [localgovdrupal/localgov_microsites#393]
Also see: https://github.com/localgovdrupal/docs/pull/201

**Comment by tonypaulbarker** [comment]
This was raised in slack and https://www.drupal.org/project/domain_simple_sitemap looks to be the solution so that each site can have its own sitemap.

**Comment by xtianjohnson** [comment]
Hi guys,

As per our discussions in the meeting, my colleague has already tried simple sitemaps.

Let us know if you have any further suggestions.


Summary of what I tried:
Simple Sitemap / XML Sitemap: only generates for the main domain, even when accessing subdomains.
Domain Simple Sitemap + Simple Sitemap: this can generate per-domain sitemaps, but:
If I don’t rebuild permissions, the sitemap jobs stay in pending state.
If I do rebuild permissions, new content shows up fine, but old content is not visible to assigned users (even after granting all likely permissions). Old nodes exist but don’t appear for the microsite users.
Is there a safe way to regenerate node/domain access so old content becomes visible again (without breaking permissions/visibility), or any steps to restore domain assignments after a rebuild?
Alternatives: should we switch to path-based sitemaps (xmlsitemap variants / custom sitemap per path) or write a small custom sitemap generator for each microsite? Any recommendations?

**Comment by ekes** [comment]
I've had a quick look at this. It does seem Domain Simple Sitemap is a little tied up with Domain Access (specifically, rather than Domain generally - ie. the Node Access model).

I could see two ways forward. Unpick the Domain Simple Sitemap a bit (there are a couple of issues in its queue). 

OR

Create a Group Sites Simple Sitemap: generate a simple sitemap variant per group site automatically (the variant is a config entity); even add it to the site group (probably would make it possible to configure it per site in the site admin); override the generator to set the site context when using that variant (possibly also create a simple_sitemap.settings per site, and so set the default); override the controller to add the group site to the cache context (and if the default hasn't been changed per last bullet, set the variant based on the active group site).

The later looks like more work, because I've kinda run through what steps are really needed, so probably isn't, I'm not so sure how much unpicking is needed for the first option. But maybe someone has already made a work-around there... in which case it would be quicker ;-)

**Comment by rifas-pbi** [comment]
Yes the problem with domain access is its require rebuilding of permission and once it is rebuild every content gets disappear from the editor content page, secondly instead they will have a domain access field across every node to select to which domain this node has to point, this also not possible since content editor of site 1 can choose the content for site 2.

**Comment by ekes** [comment]
> Yes the problem with domain access is its require rebuilding of permission

Also we don't use _Domain Access_. It's a _Node Access_ based module. The access control Microsites uses is _Entity Access_ based, using the variation cache, via _Group_ module and _Group Sites_. This means not only nodes, but other entities, can be restricted to a site. It also means _Node Access_ is not what is deciding what is on each site. **We do use** _Domain_ module for controlling which domain/url/site is active. This is then used by _Group Context: Domain_ to activate the correct _Group Site_. So the model used by _Domain Simple Sitemap_ **could** work if it was just creating a Simple Sitemap _'variant_' per domain and activating that domain when the sitemap is being generated (be that via the UI - from any other domain; or via drush; or via cron). That is however where I start to wonder if it's not as straightforward to make a _variant_ per _group_, and activate the _group site_.

**Comment by rifas-pbi** [comment]
Thanks @ekes for letting me understand the architecture of Microsites especially how we use domain module and group context. I will try what you suggested.

**Comment by rifas-pbi** [comment]
It worked. Thanks.
